### DISCUSSION TITLE
Radial Build

### DISCUSSION THREAD

**Discussion by jonmaddock** [ukaea/PROCESS#3312] [Ideas]
In GitLab by @stuartmuldrew on Oct 25, 2019, 17:25

We calculate the distance up to a component in the radial build at multiple points in the code.  Each time we just list everything in the calculation and there is a high risk of an error being introduced if we make changes to the order.

As discussed with @jonmaddock and @skahn, there should be reference for the positions, either with a function call or a set of variables, so this calculation is only done once.

Ref: @jmorris-uk

**Comment by chris-ashe** [comment]
Migrated here from an issue as this is a larger issue.

Since radial positions are calculated both absolutely and relatively at several different parts of the code and within different modules, this seems like a larger issue. The presence of `tf_in_cs` aswell introduces a build change between the TF and CS.

This may be better to implement when the code is fully Pythonised and a concurrent data structure can be set up.
